### DISCUSSION TITLE
Seeking Advice on Improving Efficiency and Overcoming WAF Challenges in Subdomain Scanning

### DISCUSSION THREAD

**Discussion by nobbieboy1337** [projectdiscovery/nuclei#4751] [Q&A]
Hi Everyone,

I hope this message finds you well.

I've been conducting subdomain scanning using Nuclei, aiming to analyze around 1000 subdomains at a time. However, I've noticed that the process is taking considerably longer than expected, and more importantly, our VPS server frequently gets stuck during the scans. This issue seems to be exacerbated by advanced Web Application Firewalls (WAFs) like Akamai, Cloudflare, and others.

I'm reaching out to the Project Discovery team and the community to seek advice on the best strategies to overcome these challenges. Specifically, I'm interested in learning:

1. How can we optimize the subdomain scanning process to enhance its speed and efficiency?
2. What are the recommended techniques to bypass or mitigate the impact of advanced WAFs during scanning?
3. How can we ensure that all Nuclei templates are effectively applied during the scanning process, even when encountering diverse domains and WAF protections?

Any insights, tips, or best practices you can provide would be greatly appreciated. Our goal is to streamline our scanning operations while ensuring comprehensive coverage and avoiding disruptions caused by WAFs.

Thank you for your time and assistance.

Best regards,
Akram

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
